docs/source/en/chat_templating.md

```diff
@@ -233,6 +233,175 @@ The sun.</s>
 
 From here, just continue training like you would with a standard language modelling task, using the `formatted_chat` column.
 
+## Can I pass other arguments to the chat template?
+
+Yes, you can! The only argument that `apply_chat_template` requires is `messages`. However, you can pass any keyword
+argument to `apply_chat_template` and it will be accessible inside the template. This gives you a lot of freedom to use
+chat templates for many things. There are no restrictions on the names or the format of these arguments - you can pass
+strings, lists, dicts or whatever else you want. 
+
+That said, there are some common use-cases for these extra arguments,
+such as passing tools for function calling, or documents for retrieval-augmented generation. In these common cases,
+we have some opinionated recommendations about what the names and formats of these arguments should be.
+
+### Arguments for tool use
+
+"Tool use" LLMs can choose to call functions as external tools before generating an answer. Our recommendation for
+tool use models is that their template
+should accept a `tools` argument. This should be a list of tools, defined via [JSON Schema](https://json-schema.org/). Each "tool"
+is a single function that the model can choose to call, and the schema should include the function name, its description
+and the expected spec for its arguments.
+
+#### Example
+
+```python
+# A simple function that takes no arguments
+current_time = {
+    "name": "current_time",
+    "description": "Get the current local time as a string.",
+    "parameters": {
+                'type': 'object',
```

Comment:
"object" has a specific meaning in JSON - it really translates to a Python `dict` rather than a Python `object`. See the [JSON Schema type equivalence list](https://json-schema.org/understanding-json-schema/reference/type)